src/lib/transaction_snark/transaction_snark.ml

```diff
@@ -458,7 +458,38 @@ module Base = struct
          in
          Boolean.Assert.is_true correct_coinbase_stack)
     in
+    let account_creation_amount_var =
+      Amount.Checked.of_fee
+        Fee.(var_of_t Coda_compile_config.account_creation_fee)
+    in
+    let%bind receiver =
+      (* A stake delegation only uses the sender *)
+      Public_key.Compressed.Checked.if_ is_stake_delegation
+        ~then_:sender_compressed ~else_:payload.body.public_key
+    in
+    (* we explicitly set the public_key because it could be zero if the account is new *)
     let%bind root =
```

Comment:
is this `root` used or is it shadowed by the below one? Should we set this to `let%bind _something`

Reply:
It is used to constraint the ledger when updating the sender account. i'll rename it.